sdk/keyvault/azure-security-keyvault/azure/security/keyvault/aio/_internal.py

```diff
@@ -3,25 +3,109 @@
 # --------------------------------------------------------------------------
-from typing import Any, Callable
+from typing import Any, Callable, Mapping, Optional, TYPE_CHECKING
+if TYPE_CHECKING:
+    try:
+        from azure.core.credentials import SupportsGetToken
+    except ImportError:
+        # SupportsGetToken is a typing_extensions.Protocol; we don't depend on that package
+        pass
 
 from azure.core.async_paging import AsyncPagedMixin
+from azure.core.configuration import Configuration
+from azure.core.pipeline import AsyncPipeline
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
+from azure.core.pipeline.transport import AsyncioRequestsTransport, HttpTransport
 from msrest.serialization import Model
 
+from .._generated import KeyVaultClient
+from .._internal import KEY_VAULT_SCOPES
+
 
 class AsyncPagingAdapter:
     """For each item in an AsyncPagedMixin, returns the result of applying fn to that item.
     Python 3.6 added syntax that could replace this (yield within async for)."""
-    def __init__(self, pages, fn):
-        # type: (AsyncPagedMixin, Callable[[Model], Any]) -> None
+
+    def __init__(self, pages: AsyncPagedMixin, fn: Callable[[Model], Any]) -> None:
         self._pages = pages
         self._fn = fn
 
     def __aiter__(self):
         return self
 
-    async def __anext__(self):
+    async def __anext__(self) -> Any:
         item = await self._pages.__anext__()
         if not item:
             raise StopAsyncIteration
         return self._fn(item)
+
+
+class _AsyncKeyVaultClientBase:
+    """
+    :param credential:  A credential or credential provider which can be used to authenticate to the vault,
+        a ValueError will be raised if the entity is not provided
+    :type credential: azure.authentication.Credential or azure.authentication.CredentialProvider
+    :param str vault_url: The url of the vault to which the client will connect,
+        a ValueError will be raised if the entity is not provided
+    :param ~azure.core.configuration.Configuration config:  The configuration for the SecretClient
+    """
+
+    @staticmethod
+    def create_config(
+        credential: "SupportsGetToken", api_version: str = None, **kwargs: Mapping[str, Any]
+    ) -> Configuration:
+        if api_version is None:
+            api_version = KeyVaultClient.DEFAULT_API_VERSION
+        config = KeyVaultClient.get_configuration_class(api_version, aio=True)(credential, **kwargs)
+        config.authentication_policy = AsyncBearerTokenCredentialPolicy(credential, scopes=KEY_VAULT_SCOPES)
+        return config
+
+    def __init__(
+        self,
+        vault_url: str,
+        credential: "SupportsGetToken",
+        config: Configuration = None,
+        transport: HttpTransport = None,
+        api_version: str = None,
+        **kwargs: Any
+    ) -> None:
+        if not credential:
+            raise ValueError("credential should be a credential object from azure-identity")
```

Comment:
should this error messaging be 'credential should be a credential object implementing supporting the TokenCredential protocol, such as those available in azure-identity'?